Source/Core/VideoCommon/CPUCull.cpp

```diff
@@ -0,0 +1,168 @@
+#else
+  return CPUCull_Scalar::TransformVertices<PositionHas3Elems, PerVertexPosMtx>;
+#endif
+}
+
+template <OpcodeDecoder::Primitive Primitive, CullMode Mode>
+static CPUCull::CullFunction GetCullFunction0()
+{
+#if defined(USE_SSE)
+  if (cpu_info.bAVX)
+    return CPUCull_AVX::CullVertices<Primitive, Mode>;
+  else
+    return CPUCull_SSE::CullVertices<Primitive, Mode>;
+#elif defined(USE_NEON)
+  return CPUCull_NEON::CullVertices<Primitive, Mode>;
+#else
+  return CPUCull_Scalar::CullVertices<Primitive, Mode>;
+#endif
+}
+
+template <OpcodeDecoder::Primitive Primitive>
+static Common::EnumMap<CPUCull::CullFunction, CullMode::All> GetCullFunction1()
+{
+  return {
+      GetCullFunction0<Primitive, CullMode::None>(),
+      GetCullFunction0<Primitive, CullMode::Back>(),
+      GetCullFunction0<Primitive, CullMode::Front>(),
+      GetCullFunction0<Primitive, CullMode::All>(),
+  };
+}
+
+CPUCull::~CPUCull() = default;
+
+void CPUCull::Init()
+{
+  m_transform_table[false][false] = GetTransformFunction<false, false>();
+  m_transform_table[false][true] = GetTransformFunction<false, true>();
+  m_transform_table[true][false] = GetTransformFunction<true, false>();
+  m_transform_table[true][true] = GetTransformFunction<true, true>();
+  using Prim = OpcodeDecoder::Primitive;
+  m_cull_table[Prim::GX_DRAW_QUADS] = GetCullFunction1<Prim::GX_DRAW_QUADS>();
+  m_cull_table[Prim::GX_DRAW_QUADS_2] = GetCullFunction1<Prim::GX_DRAW_QUADS>();
+  m_cull_table[Prim::GX_DRAW_TRIANGLES] = GetCullFunction1<Prim::GX_DRAW_TRIANGLES>();
+  m_cull_table[Prim::GX_DRAW_TRIANGLE_STRIP] = GetCullFunction1<Prim::GX_DRAW_TRIANGLE_STRIP>();
+  m_cull_table[Prim::GX_DRAW_TRIANGLE_FAN] = GetCullFunction1<Prim::GX_DRAW_TRIANGLE_FAN>();
+}
+
+u32 CPUCull::RunVertices(VertexLoaderBase* loader, OpcodeDecoder::Primitive primitive,
+                         DataReader src, u32 count)
+{
+  ASSERT_MSG(VIDEO, primitive < OpcodeDecoder::Primitive::GX_DRAW_LINES,
+             "CPUCull should not be called on lines or points");
+  const u32 stride = loader->m_native_vtx_decl.stride;
+  const bool posHas3Elems = loader->m_native_vtx_decl.position.components >= 3;
+  const bool perVertexPosMtx = loader->m_native_vtx_decl.posmtx.enable;
+  const u32 input_buffer_needed = stride * count;
+  if (m_input_buffer_size < input_buffer_needed)
+  {
+    u32 new_size = std::max<u32>(m_input_buffer_size, 1024);
+    while (new_size < input_buffer_needed)
+      new_size <<= 1;
+    m_input_buffer_size = new_size;
+    m_input_buffer.reset(static_cast<u8*>(Common::AllocateAlignedMemory(new_size, 16)));
+  }
+  if (m_transform_buffer_size < count)
+  {
+    u32 new_size = std::max<u32>(m_transform_buffer_size, 64);
+    while (new_size < count)
+      new_size <<= 1;
+    m_transform_buffer_size = new_size;
+    m_transform_buffer.reset(
+        static_cast<TransformedVertex*>(Common::AllocateAlignedMemory(new_size * 8, 16)));
+  }
+
+  VertexShaderManager::SetProjectionMatrix();
```

Comment:
The point of this isn't obvious.  Does `RunVertices` somehow use the projection matrix?  If so, I'd suggest adding a comment.